src/pykeen/sampling/negative_sampler.py

```diff
@@ -25,14 +25,21 @@ def __init__(
         self,
         triples_factory: TriplesFactory,
         num_negs_per_pos: Optional[int] = None,
+        filtered: bool = False,
     ) -> None:
         """Initialize the negative sampler with the given entities.
 
         :param triples_factory: The factory holding the triples to sample from
         :param num_negs_per_pos: Number of negative samples to make per positive triple. Defaults to 1.
+        :param filtered: Whether proposed corrupted triples that are in the training data should be filtered.
+            Defaults to False. See explanation in :func:`filter_negative_triples` for why this is
+            a reasonable default.
         """
         self.triples_factory = triples_factory
         self.num_negs_per_pos = num_negs_per_pos if num_negs_per_pos is not None else 1
+        self.filtered = filtered
+        # Tracking whether required init steps for negative sample filtering are performed
+        self._filter_init = False
```

Comment:
Instead of an additional variable use
```python
self.mapped_triples = None
```
and check for whether lazy initialization is necessary by
```python
 # Make sure the mapped triples are on the right device
if self.mapped_triples is None:
    # Copy the mapped triples to the device for efficient filtering
    self.mapped_triples = self.triples_factory.mapped_triples.to(negative_batch.device)
    self._filter_init = True
```